sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2525,17 +2526,21 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
     var properties = ctx.tablePropertyList.asScala.headOption
       .map(visitPropertyKeyValues)
       .getOrElse(Map.empty)
-    Option(ctx.comment).map(string).map {
-      properties += SupportsNamespaces.PROP_COMMENT -> _
+
+    if (properties.keySet.intersect(RESERVED_PROPERTIES.asScala.toSet).nonEmpty) {
+      throw new ParseException(s"Disallow to specify reserved properties, including" +
+        s" ${RESERVED_PROPERTIES.asScala.mkString("(", ",", ")")}.", ctx)
+    }
+
+    Option(ctx.comment).map(string).foreach {
+      properties += PROP_COMMENT -> _
     }
-    ctx.locationSpec.asScala.headOption.map(visitLocationSpec).map {
-      properties += SupportsNamespaces.PROP_LOCATION -> _
+    ctx.locationSpec.asScala.headOption.map(visitLocationSpec).foreach {
```

Comment:
nit: why `ctx.locationSpec` is a list but `ctx.comment` isn't?

Reply:
`locationSpec` is different from `comment`, shall we create a `commentSpec` as `locationSpec`

Reply:
let's leave it as it is for now